storage/src/main/java/org/apache/uniffle/storage/factory/ShuffleHandlerFactory.java

```diff
@@ -116,14 +119,21 @@ public ClientReadHandler createSingleReplicaClientReadHandler(CreateShuffleReadH
   private ClientReadHandler getMemoryClientReadHandler(CreateShuffleReadHandlerRequest request, ShuffleServerInfo ssi) {
     ShuffleServerClient shuffleServerClient = ShuffleServerClientFactory.getInstance().getShuffleServerClient(
         ClientType.GRPC.name(), ssi);
+    Roaring64NavigableMap expectTaskIds;
+    if (request.isExpectedTaskIdsBitmapFilterEnable()) {
+      Roaring64NavigableMap realExceptBlockIds = RssUtils.cloneBitMap(request.getExpectBlockIds());
+      realExceptBlockIds.xor(request.getProcessBlockIds());
+      expectTaskIds = RssUtils.generateTaskIdBitMap(realExceptBlockIds, request.getIdHelper());
+    } else {
+      expectTaskIds = request.getExpectTaskIds();
```

Comment:
Why should `epectTaskIds` be `request#getExpectTaskIds()` instead of `null`?

Reply:
My fault